### DISCUSSION TITLE
[Projects Beta] Milestones Across Repositories

### DISCUSSION THREAD

**Discussion by mitchelsellers** [community/community#6296] [Projects and Issues]
To fully manage projects, across repositories the concept of a "Shared Milestone" would be far easier to manage.  Consider the situation where you have a project that is comprised of multiple repositories, but you are working towards a common target (Q4 2021 Release) for example.  

It would be ideal to be able to set a milestone at the organization level, and use it within each of these so that the board could show the milestone column and then you group the board by those items.

**Comment by OldManFroggy** [comment]
I second this, right now I have to create milestones across repositories which is frustrating and prone to error. and a pretty common use case.

**Reply by ArthurTschinkel25** [reply]
yes bro

**Comment by kaishin** [comment]
This must be the aspect I am struggling the most with. For now I have duplicated the same milestones across several repositories but that completely beats the purpose.

**Reply by lucianobargmann** [reply]
Same here. Frustrating and beats on my head: _why haven't I seen this issue before moving my project management to Github_? 

And as we are on this, I would request the same for **custom labels**.

**Reply by jeremydiba** [reply]
Agree on the labels

**Reply by mubshrx** [reply]
agree for labels

**Reply by ignacio-borrajo** [reply]
Agree on the milestones and labels

**Reply by isanchez-ng** [reply]
Agree on the milestones and labels!

**Reply by ArthurTschinkel25** [reply]
agree on the labels

**Comment by EmiliaH** [comment]
I also agree with this suggestion! I would really like to be able to assign issues from different repositories to the same milestone!

For example we use submodules in a parent repo, and thus we need attach an issue in the submodule to a milestone in the parent repo.

**Reply by ignacio-borrajo** [reply]
We also use submodules and the day I realized about the milestones, regret to move all our repos to GitHub

**Comment by HHK1** [comment]
Is this item in the roadmap going to help solve this issue? https://github.com/github/roadmap/issues/278

**Reply by SpoonOfDoom** [reply]
> 
> 
> Is this item in the roadmap going to help solve this issue? [github/roadmap#278](https://github.com/github/roadmap/issues/278)

I don't think so, unless I'm misunderstanding it. It still would still mean that I have to duplicate a milestone across all repositories that are involved in a project.

**Reply by tleef** [reply]
This is the one you want https://github.com/github/roadmap/issues/276

**Comment by karlhorky** [comment]
I just came here to also offer this as feedback - I would like project-wide (cross-repo) milestones for tracking deadlines and product launches that deal with code in multiple repos.

**Comment by karlhorky** [comment]
If you just want a column, I realized it's possible to add a column (eg. called "Project Milestone" or "Global Milestone") in the table view now (choose a "Single Select" type):

![Screen Shot 2021-11-27 at 16 34 02](https://user-images.githubusercontent.com/1935696/143687550-4a6dccbe-daca-4d3f-8cbd-4ba7a28071a4.png)

However, I don't think you can get reports about how many issues are done vs how many open, like regular milestones can show.

**Reply by karlhorky** [reply]
Just implemented this, and it gets us most of the way to what we want, so far!

**Reply by brylie** [reply]
Yes, I'm taking this approach as a stop-gap as well.

**Reply by shaupt131** [reply]
Hmm, interesting approach.  Thanks for the suggestion, going to try this out in the meantime.

**Reply by ignacio-borrajo** [reply]
Yes but no, it is a patch to a real problem I think, no due date, no reports on how many issues are open and close, ...

**Comment by thclark** [comment]
+1000000. We're desperate for cross-repository milestones - which seem like such a simple thing they should have been done years ago. It's very difficult to implement something satisfactory without multiple actions and third party plugins - and even then there's nothing (even ZenHub) that solves the problem.

Without that, Projects Beta really isn't any more powerful (as far as I can see) than bookmarking URLs of regular projects with different filters activated. With it, Projects beta would become a self-sufficient (although admittedly lightweight) project management tool.

**Comment by markborges** [comment]
I also think this is a great idea. It seems most of us are using Milestones for controlling Releases, and if a Project can support multiple repositories, then it should definitely support common Milestones among them.

**Comment by terracoda** [comment]
Definitely need milestones that cross repositories.

**Comment by morpheus7CS** [comment]
Hi,

we have recently started using Github Projects (Beta). Our aim is to move the entire issue tracking and roadmap and development work management to Github, where our code resides. We love many of the features, but we also have some additional questions that will help us in making this change possible.

Milestones on the organizational level would help us a lot since we are dealing with 3 repository setup (API repository, client app repository, and admin app repository) and many of the changes touch at least two if not all three.

Please implement this soon as it would definitely bring the offering to a whole new level.

Kind regards,
g

**Reply by krs13** [reply]
Exactly what we're doing - we have multiple repos for different areas of our platform and I would like to tie the issues together in Releases, which is something that cross-repo milestones would accomplish.

**Comment by brylie** [comment]
As an aside, I think GitHub's lack of cross-repository project management features has been one of the primary factors driving the monorepo trend.

**Reply by thclark** [reply]
This is very insightful. I hadn't thought of that. But being presently in the middle of a difficult extraction of a monorepo into more sensible pattern, I look back and think yes - that was one of the driving factors in my earlier decision.

**Reply by OldManFroggy** [reply]
yeah, I moved from a mono repo once I hit 50+ microservices in different languages, getting things to work dev wise on an isolated microservices cost soo much time and training with developers and tooling it was just not worth it in the end.  (mainly due to GO and how it deals with modules, the same for D, and to a lessor case RUST)

but for good high-level product/team/work management milestones across repo's would make GitHub issues one of the top if not the top tool for managing code/issues of complex projects.  (really pains me, that I have to use another issue tool and only get pull updates in it, and if someone does a GitHub issue on a repo, it needs 2 way integrations (most svck) or you need to manually sync and move things...

 Same goes for a good 20 or so companies I have worked with in the past couple years... all are paying more for another tool, due to this single issue. (inc. myself...)

**Reply by KnutResys** [reply]
Agreed, we are pushing towards monorepos now much for this reason

**Comment by kalanchoej** [comment]
Also looking for a project based / cross-repo milestone solution.

**Comment by ericnelson** [comment]
+1 to this feature. Need it ASAP

**Comment by JonBushby77** [comment]
+1 Setting up milestones and labels, etc in every repository isn't ideal.

**Comment by ghost** [comment]
FYI: It’s on GitHub’s public roadmap as https://github.com/github/roadmap/issues/276, currently planned in `Future`. See [Issues-view](https://github.com/orgs/github/projects/4247/views/7)

Though it already got a lot of traction since 20**16**! on https://github.com/dear-github/dear-github/issues/93. Actually the [most 👍🏿  up-votes (on open issues)](https://github.com/dear-github/dear-github/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc):
<img width="890" alt="image" src="https://user-images.githubusercontent.com/23215416/179020059-0e196d84-b5d7-43d9-ada8-544705759f59.png">

## Duplicates
- ["Milestones should live at the project level" (#8073)](https://github.com/orgs/community/discussions/8073)

**Reply by gabriel-r** [reply]
@mitchelsellers you may want to add links/answer from @DominikGM to your original post so people are encouraged to vote up on the right places.

**Comment by lucianobargmann** [comment]
Thanks for the info, it's a shame it is closed for comments and voting up.

**Comment by hardiksondagar** [comment]
We recently moved from Asana to Github Projects, but we are using multiple repos so creating sprints across repos was painful. I did some hacking using Github API to create milestones/sprints in multiple repos at once.

Checkout here - https://hardiksondagar.github.io/cross-repository-milestones/
It requires personal access token of Github which is very easy to get.

Note - I'm not storing/stealing any kind of token on my end. It's pure frontend api calls. Do checkout code and feel free to modify it.

link - https://github.com/dear-github/dear-github/issues/93#issuecomment-867693654

**Comment by ITler** [comment]
Having milestones (and labels) across repositories could become a key feature for competing to tools like Jira. I would be able to go away from using Jira if this feature would be available.

**Comment by shaupt131** [comment]
+1.  This is a critical piece of project management functionality that is currently missing from Github Projects.

**Comment by NotNoah** [comment]
Please add GitHub  🙏🏻

**Comment by ConJabi** [comment]
+1

**Comment by fran-moreno** [comment]
Yes to both cross repo Milestones and Tags!

**Comment by akdasa** [comment]
Need it 👍

**Comment by phish108** [comment]
It is very strange that this has not been implemented together with the org-level Projects.

**Comment by terracoda** [comment]
Yes to cross-repo milestones and tags.

**Reply by JonBushby77** [reply]
Can't believe this still has not been implemented. Why not open source and let us do it for you.

**Comment by MarcosRava** [comment]
+1

**Comment by erasmus** [comment]
Does anybody know if there's a public roadmap for the Projects features?

**Reply by yeongbin-jo** [reply]
https://github.com/orgs/github/projects/4247

**Comment by xanderchu** [comment]
10000+ please make this available cuz teams would have different repos to manage

**Comment by ericsmal** [comment]
I get the impression most GitHub thinking revolves around its established core entity, the repository. It would be great to have the project uplifted as first class citizen and play a major part in GitHub software delivery. As many people have stated before, it would fill in such a big puzzle piece. For the same reasoning behind milestones and labels, I would also suggest issues and wikis to be on the project level. This would avoid having to create an overarching "parent" repository for the sake of having a centralized place where you can document your total project knowledge and create tasks that are not repository specific.

**Reply by gabriel-r** [reply]
That's the idea here, if they implement the global Milestones and Labels, those poor Drafts become generic non-repository specific tickets.

**Comment by oscarmorrison** [comment]
+1

**Comment by adamkdean** [comment]
Would be a great feature

**Comment by NesManrique** [comment]
+1

**Comment by DrLoebl** [comment]
+1 - Also I've moved to the new Project View and milestones are no longer visible on the tickets.  Does anyone know how to get them back?  Currently I have a script for generating the milestones across my multiple repos and then I add tickets to those every two weeks in our sprint planning.  It's great to be able to filter the Board on the different milestones but I can't see how to do that easily now.

**Reply by DrLoebl** [reply]
Well, I can see I can type it in the filter, but it was much easier before when I could click on it.

**Comment by crimx** [comment]
Cross-repo Milestones seems to be the closest thing to JIRA Epics which we used to manage high-level progress. Almost convinced our team to switch to GitHub Projects but sadly without workaround for this we had to stick with JIRA.

**Comment by andreabadesso** [comment]
+1

**Comment by trackrx** [comment]
+1

**Comment by fshaikh-caa** [comment]
+ 1, managing the same milestone across projects is so tedious

**Comment by treeder** [comment]
You can try using a [task list](https://docs.github.com/en/issues/managing-your-tasks-with-tasklists/about-tasklists) on an issue that represents a milestone. Then add a "milestone" label to the big meta issues so you can make views that only have the "milestone" label such as on a roadmap view for instance.

**Reply by arilivigni** [reply]
I guess this could work well if you setup the label(s) at the organization level before the repositories are created otherwise you have to then create these labels in each repository.

**Reply by phish108** [reply]
This is a poor hack as it adds extra overhead to the milestone management and changes in the plan become very difficult to track. I recon that this approach is even worse than copying all milestones to all repos.

**Reply by treeder** [reply]
What's the extra overhead you're referring to? Just curious. Task lists are quite a bit less overhead than using miletones. Even just basic stuff like adding issues to milestones is a lot more overhead than task lists. In addition, you get all the nice stuff like "tracked by" labels on the project cards, "tracked by" on each of the issue details, etc. 

I'd guess that task lists will replace milestones as they are a lot more functional.

**Reply by phish108** [reply]
The overhead comes from maintaining the actual issue and the meta issue. 

For example: If the actual issues are spread across repos and the meta issues live for example at the organization level, one has to manually maintain the actual issue and the two involved meta issues.

This gives no benefit over replicating milestones if issues are not functionally related, because milestones are already labels with target dates.

**Reply by treeder** [reply]
You add the "actual issues" from any repository to the task list of the meta issue and it links it up all nicely. Paste an issue URL from any repository into a task list and you'll see what I mean. No maintenance required.

**Reply by treeder** [reply]
To be clear, this requires the new task lists which apparently still have a waiting list you need to sign up for: https://docs.github.com/en/issues/managing-your-tasks-with-tasklists/about-tasklists

**Reply by phish108** [reply]
While the new task lists look nice, within the scope of this discussion task lists are a relatively small improvement to the meta issues (aka "previous iteration of task lists") we already have. 

Despite some similarities, GitHub's repo milestones behave quite a bit different compared to task lists. Having milestones as a shared resource at org level would be a nice improvement. For labels this works already.

**Comment by treeder** [comment]
Another way to make cross-repo milestones is to create a project field like this:

![image](https://github.com/community/community/assets/75826/ef693c00-1420-4d78-9c97-1f8a1f4f2cf1)

Then you can filter and/or create milestone specific tabs:

![image](https://github.com/community/community/assets/75826/decba516-f328-4596-8c4e-5a36b8f9157a)

**Reply by dapug** [reply]
Sure, but the issue here is that it won't then appear on the issue itself in whatever repo that issue is in, nor will the corresponding issue take advantage of the Milestones feature at the issues level for that repo.

That said, this is actually the approach we are taking, and it SUCKS because it means we must manage and track the projects at the org level with this custom field as suggested, AND manually try and keep a matching milestone at the repo/issue level (which can obviously get out of sync and ruin your day very fast).

There REALLY needs to be an org/project-level management of milestones.  I'm glad to see so many up votes and activity here, but disappointed for sure to see how this still has not been addressed after soooo long, such a basic and critical need.

**Reply by treeder** [reply]
It does show up on the issue itself in whatever repo it is, as do all project fields, on the right hand side under the "Projects" area. 

![image](https://github.com/community/community/assets/75826/8c614132-8afc-4e6b-bcda-7b8e0c9f1a5e)

So you don't need to keep a matching milestone in each repo, just one for the entire project.

**Reply by dapug** [reply]
True, but that suggests the Issues Milestone feature is useless and should be pulled from the product in favor of tracking it at the project level only.

...which actually I am in favor of, and I think what why this discussion was started in the first place.

You also lose other features of Issues Milestones if you simply abandon using it, as it's not merely an attribute on an issue. This might be a fine trade off and work around, but still raises the issue that github should make official changes here.

**Reply by treeder** [reply]
I would say that's a fair assessment. Milestones should probably be removed and I'd imagine that's the trajectory GitHub is on.

**Comment by vladimir-ikryanov** [comment]
+1. This feature would enhance the capabilities of GitHub Issues, making it an invaluable tool that numerous project managers and teams would greatly appreciate.

**Comment by david-L3V3L** [comment]
+1. Need this as well!

**Comment by mikaylagarvey** [comment]
PM here, I switched away from all other project mgmt tools to use GH Projects alongside my team... and this feature is a no-brainer! Please prioritize this item for the growing number of PMs that use this awesome tool, it would make our lives sooooo much easier. 🙏🏽

**Comment by f-rakete** [comment]
+1 Need this as well

**Comment by devpawann** [comment]
+1 Need this

**Comment by f-rakete** [comment]
this is literally the highest voted thread in the projects category. oh how we'd love at least an answer

**Comment by SiMoSiMo2** [comment]
+1

**Comment by marcustut** [comment]
+1

**Comment by wdbasson** [comment]
+1

**Reply by dlrac** [reply]
+1

**Comment by adrianzlobinski** [comment]
+1 need this

**Comment by caligo-erik** [comment]
Yup.
I'm trying out github for my multi-repo project.
This feature could be the deal-breaker for me, I'd like my milestones at project-level, ideally also with an org-wide overview of all projects and their respective milestones.

**Comment by iameskild** [comment]
I would love this feature! +1

**Comment by samuelm00** [comment]
would love to see this as well

**Comment by troyfitzwater** [comment]
+1 this would be great for our roadmap repo for tracking milestones already created and being actively updated in another repo

**Comment by AndreasAntener** [comment]
this would be awesome!

**Comment by kgrouleff** [comment]
Would also appreciate this feature! +1

**Comment by f-rakete** [comment]
I'll throw another +1 in

**Comment by Alex-Golovin** [comment]
+1

**Comment by hcansiz** [comment]
+1

**Comment by Noizept** [comment]
+1

**Comment by KnutResys** [comment]
+1

**Comment by nubowski** [comment]
+

**Comment by glebignatieff** [comment]
+1

**Comment by f-rakete** [comment]
+1

**Comment by hilakashai** [comment]
+1

**Comment by treeder** [comment]
Projects and [task lists](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/about-task-lists) will likely replace milestones.  

And can we please stop adding +1 comments so we're not all getting spammed?  Just upvote the original post.

**Reply by SydAirey** [reply]
Curious why you think that projects and task lists will replace milestones.  I have a bunch of engineers that work in different project boards so that doesnt fix the milestone feature request.

**Reply by treeder** [reply]
Couple reasons: 1) GitHub spends lot of time on projects and task lists and zero time on milestones and 2) projects and task lists can be used to track milestones (and much more) across repositories and even across organizations.

**Reply by SydAirey** [reply]
We have over 300 repos that we work in so I think its challenging to use Projects and tasks lists as milestone trackers.  I'd be curious to know how you are currently doing that for your teams.  I haven't found an easy way and everything I am tracking milestone wise is a manual effort.

**Reply by treeder** [reply]
I wrote up a little example for two different ways you can do this, with screenshots, here: https://thingster.app/things/hMX-hJkoKhtK4Te0UbcfS

**Reply by SydAirey** [reply]
Thank you so much for sharing this info! Really appreciate your help.

**Reply by SydAirey** [reply]
Hey Travis!  i finally got around to looking at this a little more and I think its a good temporary work around but heres my challenge.  We are currently re-building a food and beverage app from scratch and are working with a lot of legacy code that we inherited from another team (long story).  While we have a high level milestone identified, its nearly impossible to gather all the requirements bc we are taking more of an agile/iterative approach and learning as we progress.  I like the one project, all milestones option the best vs a project per milestone but that doesnt really help me with creating the roadmap and showing the complexity of this project.

**Reply by treeder** [reply]
Sounds like you've got some pretty complicated requirements. For people that are just wanting the old, almost featureless, milestone feature to work across repos, using projects like I said is a good way to do it.

**Comment by treeder** [comment]
For anyone wondering how they can use projects for cross repo (and org) milestones, see this thread for two options: https://thingster.app/things/hMX-hJkoKhtK4Te0UbcfS

**Comment by chmajidnaeem** [comment]


**Reply by chmajidnaeem** [reply]
Answer

**Comment by mbtre** [comment]
+1

**Comment by mawkee** [comment]
+1

**Comment by whereandy** [comment]
+1

**Comment by giosueDelgado** [comment]
+100 need it asap..

This is the related issue: https://github.com/github/roadmap/issues/276

**Comment by ewilazarus** [comment]
+1

**Comment by victorvhpg** [comment]
+1

**Comment by ghost** [comment]
Do we need to get our pitchforks?  +1

**Comment by jimrazmus** [comment]
+1 Setting up a consistent set of milestones, status, and labels in numerous repositories is extremely cumbersome.

**Comment by treeder** [comment]
Do any of you have access to the new GitHub projects?  I think you wouldn't be complaining about this feature if you did.

**Reply by CdrMarks** [reply]
@treeder I do. I think they call the older offering Projects (classic).

I'm curious what object you use for the milestone?

**Reply by treeder** [reply]
Yes, org level projects: https://github.com/features/issues  

There's various ways to organize it, you could have one project per milestone, or one project and have a column per milestone, or one project and one issue per milestone with a "task list" in the issues with all the milestone tasks. 

This new stuff is 10x better than the old milestones.

**Reply by CdrMarks** [reply]
Here's a use case I'd like to implement. 

I have the following set up:
* Issues from several projects can be seen in one org level project named Scrum.  
  * The custom fields in that project are currently: status, sprint, and story points.
  * Various views used to plan and execute sprints
* Issues from a repository named roadmap in a separate org level project named Roadmap.
  * The custom fields in that project are currently: quarter
  * In each roadmap issue, I include a bulleted list of the issues from the other projects (I'm using these like epics)

An idea I have is the following:
* In the Scrum project, add a custom field of type Single select called Epic.  
* For each epic give it a short name that is based on the issue name from the roadmap repository.  Also add in the epic's description the URL of the roadmap issue.
* In each roadmap issue, include a link to the scrum project using a view filtered to only show this epic's issues.

I'm curious how what I described above can be achieved in one project.

**Reply by treeder** [reply]
You can easily filter views by that epic field and links can have the filters in them: 
![image](https://github.com/community/community/assets/75826/0dbd7841-31ea-4354-aa06-df8e263ae72e)

If it all has to be in one project, you could also label the main roadmap issues as "roadmap" then filter those out for a roadmap view (there's even a view type called roadmap that's like a Gantt chart). 

Best thing to do is just try it and you'll probably figure out a way to do what you want.

**Reply by treeder** [reply]
![image](https://github.com/community/community/assets/75826/4c9edc10-3835-4ceb-b628-2aabc757b4f5)

**Comment by DeveloperAlly** [comment]
+100000000 Do we have any word on if they have scheduled this at least?

I guess we better not hold our breath - people have been asking for almost 10 years -> https://github.com/dear-github/dear-github/issues/93

**Reply by caligo-erik** [reply]
Not holding my breath, but I still hope for this feature.

**Comment by luisarojas** [comment]
+1

**Comment by djcesar** [comment]
+1

**Comment by gparker** [comment]
+1

**Comment by mtfunder** [comment]
+1

**Comment by imricardosilva** [comment]
+1

**Comment by AgnesAndersson** [comment]
+1

**Comment by junkim012** [comment]
please!

**Comment by m00nl1ghtP00l** [comment]
+1

**Comment by AmitKKhanchandani** [comment]
+1

**Comment by Mayurbarve** [comment]
+1 Setting up a consistent set of milestones, status, and labels in numerous repositories is extremely cumbersome.

**Comment by rgbsys** [comment]
+1

**Comment by rambech** [comment]
+1

**Comment by Olshansk** [comment]
One of the latest updates for those following along:

![Screenshot 2024-12-03 at 2 05 25 PM](https://github.com/user-attachments/assets/30364a75-1d09-4cd9-8d3c-e31aa15d8b75)

https://github.com/github/roadmap/issues/276

**Comment by m-araujo-voltalia** [comment]
+1

**Comment by four43** [comment]
Seems like they're tracking here as future, but no plans to implement yet: https://github.com/github/roadmap/issues/1086

**Comment by quantuminformation** [comment]
missing this feature already

**Comment by YamilaChan** [comment]
We need milestones that cross repositories.

**Comment by batatop** [comment]
When creating a milestone for a feature, needing to do the same thing for both the app and the server repos feels tedious.

**Reply by caligo-erik** [reply]
Yup, plus you might even have shared libraries/repos.

**Comment by jmackie** [comment]
+1

**Comment by alexanderilyin** [comment]
This is the only reason why I'm considering paying for Jira. Please don't make me do that.

P. S. Cloud Jira is nice but really there is no need to have it in addition to GH.

**Reply by VeCard** [reply]
> This is the only reason why I'm considering paying for Jira. Please don't make me do that.
> 
> P. S. Cloud Jira is nice but really there is no need to have it in addition to GH.

Only that? Really? And what about the fact that there are no issues that work across repositories. That is pretty dumb in 2025. So if I want an Epic that has children user stories or tasks or whatever kind of issue you want u don't have a cross container.
That is just another example.

**Comment by Norbiros** [comment]
I personally use `Iteration` field in Github Projects. It works almost identically, but I can view it in projects and it works between multiple repositories. The only drawback for me is that I can't filter easily in Issues page (I have to open project)

**Reply by maho-jared** [reply]
This is a super interesting workaround! Open the issue in the milestone and you can pick a different repository than the one the milestone was created in. Thanks for sharing!

**Reply by tomasMizera** [reply]
We use this approach too, it works very well 👍🏻

**Reply by SydAirey** [reply]
do you use 2 week iterations? How do you think this is related to tracking milestones or issues across multiple repos that support a specific milestone or objective?

**Reply by Norbiros** [reply]
You could use x-week fixed iterations as normal iterations, and that's okay. I am using them more like milestones:
<img width="1215" height="367" alt="image" src="https://github.com/user-attachments/assets/dccdb48e-db06-40de-af1e-6806f844bbd2" />

So we plan milestones, give some arbitrary deadlines (they do often shift) and then just assign tasks to specific iteration. Thanks to that in GH projects view I can filter, this "milestone", next, previous etc.

**Comment by natemcintosh** [comment]
Building off the comment above, you can create a new column in the "Table" view, but instead of `Iteration` (which operate over a span of time), you can make it a `Text` column. Then you can type in a "milestone" value for each issue in the project, and then group by the "milestone" column. (Note that it won't let you name the column "Milestone", saying that name is already taken :roll_eyes:).

As the comment above says, you won't be able to filter easily in the Issues page, only in the project page.

**Comment by aikenka** [comment]
Added my vote for this - I'll use the workaround described in the last 2 comments; what's missing in that solution is the ability to provide notes/description for each.
Workaround for that - I'll create a "first issue" for each and put the notes in there.
I can't use parent/child relationships, since an issue can only have one parent, and the issues already have parents in other projects.

**Comment by EugenMayer** [comment]
Just as a reference, a public roadmap items exist for that https://github.com/github/roadmap/issues/1086 (which superseded https://github.com/github/roadmap/issues/776)

**Comment by olekristensen** [comment]
I'd love to see them focus on basic features like this. It's been part of GitLab, that I prefer, for ages, but my main customers enforce GitHub.  Imagine how fast these basic features could be released if they put as much energy into them as they do in smearing everything with copilot...

**Comment by pepinse** [comment]
Upvote!

**Comment by ranjanvasudevan** [comment]
+1!

**Comment by JorritBerendsen** [comment]
+1!

**Comment by minttraveller** [comment]
+1

**Comment by mofazzelhmorshed** [comment]
Yes, you shared Milestones transform disparate repo task lists into a coherent, measurable project roadmap, i think this feature is essential for high level visibility

**Comment by sebastianhorwege** [comment]
1

**Comment by JRiad** [comment]
Please, please, please add this feature! 🙏

**Comment by isasmendiagus** [comment]
+1

**Comment by loeffel-io** [comment]
+100

**Comment by BBaoVanC** [comment]
Please stop posting comments that are purely "+1" -- it clutters up the comment feed, sends a (somewhat useless) notification to everyone that is watching, and is more more visible if you instead just add an upvote and thumbs up reaction to the main post

**Comment by claudioLOXO** [comment]
I think that, more broadly, having clearer structure in GitHub — especially through milestones across multiple repositories — would actually help us reduce noise and improve alignment overall.

I think that right now we manage issues well inside each repo, but when a feature or release spans multiple repos, it becomes harder to maintain a clear overview. We end up mentally stitching together progress across different boards and PRs. A shared milestone gives us one explicit goal. Even if the work is distributed, the milestone represents a single outcome and makes it much easier to see what’s open, what’s merged, and what might block completion.

I think that milestones also help with scope control. When we define one properly, we’re implicitly agreeing on what belongs to that delivery. If something new appears mid-way, we’re forced to consciously decide whether it fits into the current milestone or should move to the next one. That helps prevent silent scope creep and keeps timelines more realistic.

I think that cross-repo coordination would also improve. Many changes depend on updates elsewhere, and when related issues share the same milestone name, those dependencies become visible instead of implicit.

Overall, I think this isn’t about adding overhead. It’s about creating clarity and reducing friction as complexity grows.
...
...
+1

**Reply by Risbo6** [reply]
+1

**Comment by lucasrabaca** [comment]
At this point, GitHub just doesn’t want to compete in project management, because it makes no sense for projects or milestones not to have their own issues independent of a repository—or for milestones to be fragmented across repositories instead of being truly project-level.

**Comment by amurugas** [comment]
+1 need this

**Comment by otonielguajardo** [comment]
Just as a reference, here's the public roadmap item https://github.com/github/roadmap/issues/1086

**Comment by nugrahn0123** [comment]
+1
